src/Datadog.Trace/Sampling/RegexSamplingRule.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Sampling
+{
+    internal class RegexSamplingRule : ISamplingRule
+    {
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<RegexSamplingRule>();
+
+        private readonly float _samplingRate;
+        private readonly string _serviceNameRegex;
+        private readonly string _operationNameRegex;
+
+        public RegexSamplingRule(
+            float rate,
+            string name,
+            string serviceNameRegex,
+            string operationNameRegex)
+        {
+            _samplingRate = rate;
+            _serviceNameRegex = serviceNameRegex;
+            _operationNameRegex = operationNameRegex;
+
+            Name = name;
+        }
+
+        public string Name { get; }
+
+        /// <summary>
+        /// Gets the Priority of the rule.
+        /// Configuration rules will default to 1 as a priority and rely on order of specification.
+        /// </summary>
+        public int Priority => 1;
+
+        public static IEnumerable<RegexSamplingRule> BuildFromConfigurationString(string configuration)
+        {
+            if (!string.IsNullOrEmpty(configuration))
+            {
+                var ruleStrings = configuration.Split(new[] { ";", ":" }, StringSplitOptions.RemoveEmptyEntries);
+                var index = 0;
+
+                foreach (var ruleString in ruleStrings)
+                {
+                    index++;
+
+                    var ruleParts = ruleString.Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries);
+
+                    var rateSet = false;
+                    float rate = 0;
+                    string serviceNameRegex = null, operationNameRegex = null, ruleName = $"config_rule_{index}";
+
+                    foreach (var rulePart in ruleParts)
+                    {
+                        var kvp = rulePart.Split(new[] { "=" }, StringSplitOptions.None);
+
+                        if (kvp.Length != 2 || string.IsNullOrWhiteSpace(kvp[1]))
```

Comment:
Should we also check for `string.IsNullOrWhiteSpace(kvp[0])` here to catch a rule like `=1.0`, where there's nothing to the left of the `=`?
```suggestion
                        if (kvp.Length != 2 ||
                            string.IsNullOrWhiteSpace(kvp[0]) ||
                            string.IsNullOrWhiteSpace(kvp[1]))
```